tools/buildsys/src/gomod.rs

```diff
@@ -0,0 +1,213 @@
+        let local_file_name = &external_file.path.as_ref().unwrap_or(&url_file_name);
+        ensure!(
+            local_file_name.components().count() == 1,
+            error::InputFileSnafu
+        );
+
+        let full_path = package_dir.join(local_file_name);
+        ensure!(
+            full_path.is_file(),
+            error::InputFileBadSnafu { path: full_path }
+        );
+
+        // If a module directory was not provided, set as an empty path.
+        // By default, without a provided module directory, tar will be passed
+        // the first directory found in the archives as the top level Go module
+        let default_empty_path = PathBuf::from("");
+        let mod_dir = external_file
+            .bundle_root_path
+            .as_ref()
+            .unwrap_or(&default_empty_path);
+
+        // Use a default "bundle-{name-of-tar-file}" if no output path was provided
+        let default_output_path =
+            PathBuf::from(format!("bundled-{}", local_file_name.to_string_lossy()));
+        let output_path_arg = external_file
+            .bundle_output_path
+            .as_ref()
+            .unwrap_or(&default_output_path);
+        println!(
+            "cargo:rerun-if-changed={}",
+            output_path_arg.to_string_lossy()
+        );
+
+        // Our SDK and toolchain are picked by the external `cargo make` invocation.
+        let sdk = env::var("BUILDSYS_SDK_IMAGE").context(error::EnvironmentSnafu {
+            var: "BUILDSYS_SDK_IMAGE",
+        })?;
+
+        let args = DockerGoArgs {
+            module_path: package_dir,
+            sdk_image: sdk,
+            go_mod_cache: &root_dir.join(".gomodcache"),
+            command: format!("./{}", GO_MOD_DOCKER_SCRIPT_NAME),
+            extra_docker_args: vec![
+                // Sets container environment variables needed for the docker-go-script
+                format!("-e LOCAL_FILE_NAME={}", local_file_name.to_string_lossy()),
+                format!("-e MOD_DIR={}", mod_dir.to_string_lossy()),
+                format!("-e OUTPUT={}", output_path_arg.to_string_lossy()),
+            ],
+        };
+
+        // Create and/or write the temporary script file to the package directory
+        let script_path = format!(
+            "{}/{}",
+            package_dir.to_string_lossy(),
+            GO_MOD_DOCKER_SCRIPT_NAME
+        );
+        {
+            let mut script_file = fs::File::create(&script_path).unwrap();
+            fs::set_permissions(&script_path, fs::Permissions::from_mode(0o777)).unwrap();
+            script_file.write_all(GO_MOD_SCRIPT.as_bytes()).unwrap();
+        }
+
+        match docker_go(root_dir, &args) {
+            Ok(_) => {
+                fs::remove_file(script_path).unwrap();
+                Ok(())
+            }
+            Err(e) => {
+                fs::remove_file(script_path).unwrap();
+                Err(e)
+            }
+        }
```

Comment:
Hmm, a more idiomatic way to do this might be:
```suggestion
        let res = docker_go(root_dir, &args);
        fs::remove_file(script_path)?;
        res
```